backend/server-connection/src/main/java/org/sonarsource/sonarlint/core/serverconnection/AnalyzerConfigurationStorage.java

```diff
@@ -116,7 +118,10 @@ private static ServerActiveRule adapt(Sonarlint.RuleSet.ActiveRule rule) {
       rule.getRuleKey(),
       IssueSeverity.valueOf(rule.getSeverity()),
       rule.getParamsMap(),
-      rule.getTemplateKey());
+      rule.getTemplateKey(),
+      rule.getOverriddenImpactsList().stream()
```

Comment:
You probably lack the reverse operation in the other `adapt`: when writing to the storage you need to call the setter